corehq/apps/locations/middleware.py

```diff
@@ -31,6 +36,16 @@ class LocationAccessMiddleware(MiddlewareMixin):
     def process_view(self, request, view_fn, view_args, view_kwargs):
         user = getattr(request, 'couch_user', None)
         domain = getattr(request, 'domain', None)
+
+        # Initial digest requests will not have any auth information
+        # this will be handled elsewhere - we can safely ignore
+        has_auth = determine_authtype_from_header(request, default='NONE') != 'NONE'
+        if domain and has_auth:
+            # This should eventually be made into a hard requirement, but I'm
+            # not yet sure what that might break
+            _assert_user_and_domain(user, "A request was just made with a domain but no user. "
+                                          "Is this normal?  How can we authenticate it?")
```

Comment:
@kaapstorm do you recall why this is a soft assert rather than a notify exception?  Do soft alerts show up on sentry?  We'd definitely want a plethora of information about what happens when this is triggered?